core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -262,15 +267,22 @@ private[spark] class ExecutorAllocationManager(
     val maxNeeded = maxNumExecutorsNeeded
 
     if (maxNeeded < numExecutorsTarget) {
-      // The target number exceeds the number we actually need, so stop adding new
-      // executors and inform the cluster manager to cancel the extra pending requests
-      val oldNumExecutorsTarget = numExecutorsTarget
-      numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
-      numExecutorsTarget - oldNumExecutorsTarget
+      if (!numTargetExecutorAdjustable.get) {
+        // Keep the initial number of target executor to not ramp down until the first job is
```

Comment:
IIUC, this will stop us from adjusting our target down to something greater than initialExecutors.  E.g. if initialExecutors is 5 and we raise targetNumExecutors to 10, we won't then be able to decrease it down to 7.